test/junit/scala/collection/immutable/TreeSetTest.scala

```diff
@@ -213,4 +213,37 @@ class TreeSetTest extends AllocationTest{
       assertEquals(s"$rText $lText", rhs, lhs)
     }
   }
+
+  @Test
+  def unionAndIntersectRetainLeft(): Unit = {
+    case class C(a: Int)(override val toString: String)
+    implicit val ordering: Ordering[C] = Ordering.by(_.a)
+    val c0l = C(0)("l")
+    val c0r = C(0)("r")
+    def assertIdenticalElements(expected: Set[C], actual: Set[C]): Unit = {
+      val expected1, actual1 = Collections.newSetFromMap[C](new java.util.IdentityHashMap())
+      expected.foreach(expected1.add)
+      actual.foreach(actual1.add)
+      assertEquals(expected1, actual1)
```

Comment:
TIL
I didnt know that IdentityHashMap `.equals` was identity based
but it is and I have verified that it is